country-dashboard/src/Component/Card.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState, Dispatch, SetStateAction } from "react";
+import axios from "axios";
+import { BASE_URL } from "./api";
+import { Box, Text, Image, Spinner, AspectRatio } from "@chakra-ui/react";
+
+interface CardsProps {
+  selectedCurrency: string | null;
+  selectedLanguage: string | null;
+  selectedRegion: string | null;
+  setSearchValue: Dispatch<SetStateAction<string>>;
+  searchValue: string;
+}
+
+export const Cards: React.FC<CardsProps> = ({
+  selectedCurrency,
+  selectedLanguage,
+  selectedRegion,
+  setSearchValue,
+  searchValue,
+}) => {
+  const [filteredCountries, setFilteredCountries] = useState<any[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchData = async () => {
+    setLoading(true);
+    setError(null);
+    let url = BASE_URL + "all";
+    if (searchValue) {
+      url = BASE_URL + `name/${searchValue}`;
+    } else if (selectedCurrency) {
```

Comment:
Unfortunately, this doesn't handle AND conditions well, for example if I enter "English" for language and "Asia" for region, I get countries in Asia that don't speak English. The results should have been merged, using a current state, which contains already filtered data, and then only displaying countries that are in both `current` (already filtered) and the API result (new filter). -5

Reply:
Not taking points off for this